x-pack/plugins/ml/server/routes/data_frame_analytics.ts

```diff
@@ -575,4 +587,40 @@ export function dataFrameAnalyticsRoutes({ router, mlLicense }: RouteInitializat
       }
     })
   );
+
+  /**
+   * @apiGroup DataFrameAnalytics
+   *
+   * @api {get} /api/ml/data_frame/analytics/map/:analyticsId Get objects leading up to analytics job
+   * @apiName GetDataFrameAnalyticsIdMap
+   * @apiDescription Returns map of objects leading up to analytics job.
+   *
+   * @apiParam {String} analyticsId Analytics ID.
+   */
+  router.get(
+    {
+      path: '/api/ml/data_frame/analytics/map/{analyticsId}',
+      validate: {
+        params: analyticsIdSchema,
+        query: analyticsMapQuerySchema,
+      },
+    },
+    mlLicense.fullLicenseAPIGuard(async ({ client, request, response }) => {
+      try {
+        const { analyticsId } = request.params;
+        // @ts-ignore
+        const treatAsRoot = request.query?.treatAsRoot;
+        const caller =
+          treatAsRoot === 'true' || treatAsRoot === true ? getExtendedMap : getAnalyticsMap;
```

Comment:
`getAnalyticsMap` and `getAnalyticsMap` are fairly complex and deserve API integration tests

Reply:
Included in the meta issue for a follow up 👍